File: app/controller/builder.rb
# frozen_string_literal: true

require_relative 'source_data_reader'
require_relative '../model/competency_level'
require_relative '../view/renderer'

class Builder

  def initialize
    reader.create_master_yaml
    topics = Topic.all
    renderer = Renderer.new(topics)
    renderer.render_semesters
  end

  private

  def reader
    @reader ||= SourceDataReader.new
  end
end

File: src/main/java/controller/Builder.java
package controller;

import model.Plan;

import java.util.ArrayList;
import java.util.HashMap;

public class Builder{

    private static final String FILE_PATH_YAMLS_TO_PROCESS = "src/main/resources/data";

    public static void main(String[] args) throws Exception {
        SourceDataReader reader = new SourceDataReader();
        HashMap<String, ArrayList> mapYamlObjects = reader.readAndPrepareAllSourceDataYamls(FILE_PATH_YAMLS_TO_PROCESS);

        TopicCreator topicCreator = new TopicCreator();
        Plan plan = new Plan(topicCreator.createTopics(mapYamlObjects), reader.getSemester());
        //TODO: render plan
        reader.createMasterYaml(plan);



    }

}